consensus/istanbul/backend/announce.go

```diff
@@ -835,18 +800,43 @@ func (sb *Backend) handleVersionedEnodeMsg(peer consensus.Peer, payload []byte)
 		return err
 	}
 
-	// There may be a difference in the URLv4 string because of `discport`,
-	// so instead compare the ID
-	selfNode := sb.p2pserver.Self()
-	if parsedNode.ID() != selfNode.ID() {
-		logger.Warn("Received Istanbul Versioned Enode message with an incorrect enode url", "message enode url", versionedEnode.EnodeURL, "self enode url", sb.p2pserver.Self().URLv4())
-		return errors.New("Incorrect enode url")
+	// Handle the special case where this node is a proxy and the proxied validator
+	// sent a versioned enode for the proxy to use in handshakes
+	if sb.config.Proxy && msg.Address == sb.config.ProxiedValidatorAddress {
+		// There may be a difference in the URLv4 string because of `discport`,
+		// so instead compare the ID
+		selfNode := sb.p2pserver.Self()
+		if parsedNode.ID() != selfNode.ID() {
+			logger.Warn("Received Istanbul Versioned Enode message with an incorrect enode url", "message enode url", versionedEnode.EnodeURL, "self enode url", sb.p2pserver.Self().URLv4())
+			return errors.New("Incorrect enode url")
+		}
+		sb.setVersionedEnodeMsg(&msg)
```

Comment:
Nice catch! To avoid the work of having to decode the `istanbul.Message` contents to compare the existing version, I changed the surrounding if condition to make sure the peer is the proxied validator.